src/txn/txn_rollback_to_stable.c

```diff
@@ -201,66 +201,69 @@ __txn_abort_newer_row_leaf(
     }
 }
 
+/*
+ * __txn_page_needs_rollback --
+ *     Check whether the page needs has modification newer than the given timestamp.
+ */
+static int
+__txn_page_needs_rollback(WT_SESSION_IMPL *session, WT_REF *ref, wt_timestamp_t rollback_timestamp)
+{
+    WT_ADDR *addr;
+    WT_CELL_UNPACK vpack;
+    WT_MULTI *multi;
+    WT_PAGE_MODIFY *mod;
+    wt_timestamp_t multi_newest_durable_ts;
+    uint32_t i;
+
+    addr = ref->addr;
+    mod = ref->page == NULL ? NULL : ref->page->modify;
+
```

Comment:
Given that there are 4 different conditionals here, a top-level comment describing all the different options here would be helpful.

Reply:
Added a comment to explain the different checks that are carried out to find out whether the page needs rollback or not.